app/src/main/java/com/demoapp/ui/MainActivity.kt

```diff
@@ -6,26 +6,36 @@ import android.view.MenuItem
 import android.view.View
 import android.widget.Toast
 import androidx.lifecycle.ViewModelProvider
+import com.demoapp.App
 import com.demoapp.R
 import com.demoapp.adapter.CategoryDataAdapter
+import com.demoapp.extension.getViewModel
 import com.demoapp.viewmodel.MainActivityViewModel
 import kotlinx.android.synthetic.main.activity_main.*
+import javax.inject.Inject
 
 class MainActivity : AppCompatActivity() {
 
-    lateinit var viewModel: MainActivityViewModel
+    @Inject
+    lateinit var viewModelFactory: ViewModelProvider.Factory
+
+    private val viewModel: MainActivityViewModel by lazy {
+        getViewModel(viewModelFactory)
+    }
+
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
 
-        var categoryDataAdapter = CategoryDataAdapter(this@MainActivity)
+        var categoryDataAdapter = CategoryDataAdapter()
         recycleList.adapter = categoryDataAdapter
 
         supportActionBar?.title= ""
         supportActionBar?.setDisplayHomeAsUpEnabled(true)
 
-        viewModel = ViewModelProvider(this).get(MainActivityViewModel::class.java)
+        (application as App).getAppComponent().inject(this)
```

Comment:
Do injection as the 1st line of code in onCreate

Reply:
This change is done